foundations/src/security/mod.rs

```diff
@@ -529,7 +460,7 @@ pub fn forbid_x86_64_cpu_cycle_counter() {
 /// ```
 ///
 /// [prctl(PR_GET_SECCOMP)]: https://linuxman7.org/linux/man-pages/man2/PR_GET_SECCOMP.2const.html
-fn get_current_thread_seccomp_mode() -> BootstrapResult<SeccompMode> {
+pub fn get_current_thread_seccomp_mode() -> BootstrapResult<SeccompMode> {
```

Comment:
The example seems to be incomplete - it doesn't show the usage of the API

Reply:
We also have an unused option in `SeccompMode` which is confusing. Should we instead just have `is_seccomp_enabled_for_thread() -> BootstrapResult<bool>`?

Reply:
Good point - I'll fill out the example

Also, yeah, I agree about SeccompMode. I think I did that to try and communicate the quirk that you'll never find out that you're in strict mode, since if you call `prctl` in strict mode, your process will be killed. But I think I can do that well enough in the doc on this method and improve the API as you described, since almost every caller would probably do `matches!(seccomp_mode, SeccompMode::Filter)` or similar to convert its return type to a boolean.

I'll open a new PR addressing those.

Reply:
See #75